src/main/java/io/strimzi/kproxy/interceptor/HandlerContext.java

```diff
@@ -16,13 +16,13 @@
  */
 package io.strimzi.kproxy.interceptor;
 
-import java.nio.ByteBuffer;
+import io.netty.buffer.ByteBuf;
 
 /**
  * Provides contextual information to request and response handlers.
  */
 public interface HandlerContext {
     String channelDescriptor();
 
-    ByteBuffer allocate(int initialCapacity);
+    ByteBuf allocate(int initialCapacity);
```

Comment:
Given the class's doc of "provides contextual information is this the right place for this method?

Reply:
Agree, but let @gunnarmorling decide what to do with it :)

Reply:
It is the right place, it's not just meant for information but also as a place for "action" methods like this (we can rename it, if you think we should). But why do we expose Netty's ByteBuf here now? That's what I was hoping to avoid, so as to make handler implementations independent from the underlying network stack, if possible.

Reply:
I think the name is fine, just adjust the javadoc to broaden the abstraction so it's not just about providing information.

Reply:
> But why do we expose Netty's ByteBuf here now?

Because if such buffers are pooled, without the `ByteBuf` cannot be released in the pool